### ISSUE TITLE
How do I add more languages to the translate options?

### ISSUE THREAD

**Issue by phdumais** [amazon-connect/ai-powered-speech-analytics-for-amazon-connect#29]
Is it possible to add more languages to translate into? I'm trying to add French, but I have no idea where I might have to modify the code.

**Comment by mcosicaws** [comment]
Is that related to voice or chat? 

For voice, the solution utilize Amazon Transcribe for speech to text, therefore the language has to be supported by Amazon Transcribe. More details can be found here: 
https://docs.aws.amazon.com/transcribe/latest/dg/API_streaming_StartStreamTranscription.html
Please check `LanguageCode` parameter. 
Once you confirmed the language is supported, you need to set the `languageCode` parameter in your Contact Flow.
More details can be found here:
https://docs.aws.amazon.com/solutions/latest/ai-powered-speech-analytics-for-amazon-connect/controlling-solution-behavior.html

On the client side, customer language is then set on this line:
https://github.com/amazon-connect/ai-powered-speech-analytics-for-amazon-connect/blob/main/source/web_site/agentAssist.html#L487
`customerLanguage = fromLanguageDict[attr.languageCode.value];`

List of agent languages (drop-down selection) is defined here:
https://github.com/amazon-connect/ai-powered-speech-analytics-for-amazon-connect/blob/main/source/web_site/agentAssist.html#L1274
You can expand this list with any additional languages supported by Amazon Translate service. 

The solution utilize Amazon Translate, please check the list of supported languages here:
https://docs.aws.amazon.com/translate/latest/dg/what-is.html#language-pairs

Therefore, for voice, customer language (set in Contact Flow) is translated into the agent language (set in the drop-down list)

On the other hand, chat messages do not require Amazon Transcribe, and solution utilizes Amazon Comprehend to detect the customer language: 
https://github.com/amazon-connect/ai-powered-speech-analytics-for-amazon-connect/blob/main/source/web_site/agentAssist.html#L860

List of agent languages (drop-down selection) is defined here:
https://github.com/amazon-connect/ai-powered-speech-analytics-for-amazon-connect/blob/main/source/web_site/agentAssist.html#L1192
You can expand this list with any additional languages supported by Amazon Translate service. 

Basically, for chat, customer detected language is then translated into the selected agent language. 

I hope this helps, but please let us know if you need more details.

Thanks,
Milos

**Comment by mcosicaws** [comment]
Closing due to no response, please feel free to re-open should you need any additional help.
